go/enclave/limiters/batchlimiter.go

```diff
@@ -0,0 +1,75 @@
+package limiters
+
+import (
+	"errors"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+// BatchSizeLimiter - Acts as a limiter for batches based
+// the data from the transaction that we have to publish to the l1.
+// Acts as a calldata reservation system that accounts for both
+// transactions and cross chain messages.
+type batchSizeLimiter struct {
+	Size uint64
```

Comment:
what is this size?
why is it public?
needs a comment. 
Ideally renamed as remainingSize, the same as for the rollupLimiter